src/multi-select.js

```diff
@@ -141,17 +177,14 @@ class MultiSelect extends PureComponent {
 		if (s === SPACE) {
 			const slicedItems = hasLimit ? arrRotate(items, rotateIndex).slice(0, limit) : items;
 			const selectedItem = slicedItems[highlightedIndex];
-			const selectedItemKey = selectedItem.key || selectedItem.value;
-
-			(selected[selectedItemKey] ? onUnselect : onSelect)(selectedItem);
 
-			this.setState(state => ({
-				selected: {...selected, [selectedItemKey]: !state.selected[selectedItemKey]}
-			}));
+			this.setSelectedState(
+				this.selectItem(selectedItem)
+			);
```

Comment:
Feels like the state setting should happen in the `selectItem` method, like the name suggests. Can we move the `this.setState` from being in its own method to happen within the `selectItem`?

Reply:
Happy to move the setState method into selectItem however this setup does allow for an escape hatch when setting selected items without triggering onSelect / onUnselect events. Additionally, allows passing the entire selected items state instead of iterating on selectItem.